### ISSUE TITLE
update the sftp/ssh ciphers supported by the secureftp plugin

### ISSUE THREAD

**Issue by drzraf** [aptana/studio3#342]
In server logs:
```
client  aes128-cbc,3des-cbc,aes192-cbc,aes256-cbc,blowfish-cbc,arcfour
server aes128-ctr,aes192-ctr,aes256-ctr,aes128-gcm@openssh.com,aes256-gcm@openssh.com,chacha20-poly1305@openssh.com
```

Citing man sshd_config 6.7p1-5  (Jessie):
```
     Ciphers
             Specifies the ciphers allowed for protocol version 2.  Multiple ciphers must be comma-separated.  The supported ciphers are:

                   3des-cbc
                   aes128-cbc
                   aes192-cbc
                   aes256-cbc
                   aes128-ctr
                   aes192-ctr
                   aes256-ctr
                   aes128-gcm@openssh.com
                   aes256-gcm@openssh.com
                   arcfour
                   arcfour128
                   arcfour256
                   blowfish-cbc
                   cast128-cbc
                   chacha20-poly1305@openssh.com

             The default is:

                   aes128-ctr,aes192-ctr,aes256-ctr,
                   aes128-gcm@openssh.com,aes256-gcm@openssh.com,
                   chacha20-poly1305@openssh.com
```

From:
https://github.com/aptana/studio3/blob/0635366570ab5a90cacc1165391fc2eab90c3536/plugins/com.aptana.filesystem.secureftp/src/com/aptana/filesystem/secureftp/internal/SFTPConnectionFileManager.java
(last updated in 2012), the SSHFTPAlgorithm is inherit from:
```import com.enterprisedt.net.ftp.ssh.SSHFTPAlgorithm;```

I found it here:
https://enterprisedt.com/products/edtftpjssl/doc/api/com/enterprisedt/net/ftp/ssh/SSHFTPAlgorithm.html
which (only) defines the following ciphers:
```
CIPHER_3DES_CBC                                                                      
CIPHER_AES128_CBC                                                                    
CIPHER_AES128_CTR                                                                    
CIPHER_AES192_CBC                                                                    
CIPHER_AES192_CTR                                                                    
CIPHER_AES256_CBC                                                                    
CIPHER_AES256_CTR                                                                    
CIPHER_ARCFOUR                                                                       
CIPHER_BLOWFISH_CBC
```

But the the plugins (or aptana), in some way, removed the only common ciphers (the 128, 192 and 256) in CTR version.
Why? Where (in the builds?) ? Could you reintroduce it?

**Comment by drzraf** [comment]
In https://stackoverflow.com/questions/26577494 it was mentioned that a similar issue affects (or used to affect) the key exchange methods.

**Comment by JBSteve** [comment]
Just wanted to add on here that we are no longer able to use Aptana with our Host (WPEngine) because they have removed certain ciphers for connecting via SFTP.

Their explanation via a support exchange:

1) In one of our recent deployments we've made some updates to our SFTP compatibility. In order to ensure a secure hosting environment for all of our customers, we have removed certain cyphers from our platform. Theses cyphers are as follows : CBC, RC4, 3DES, MD5, and RIPEMD.

Two known clients that use this encryption in their software are RapidCSS, and Aptana. I would recommend reaching out to the developers of your client to request an update to the software. We apologize for any inconvenience that this may cause, but again it is to ensure stability and security for all of our customers.



2) Sorry for any confusion, what I'm suggesting is that because an industry standard client like FileZilla is working, that the issue is with Aptana itself, and as such, the Aptana support team will need to troubleshoot the issue, we can confirm SFTP is up and running and works from a standard SFTP client.

I can confirm that myself and another developer have the same inability to connect via sftp to WPEngine.

**Comment by greghouston** [comment]
Having the same issue as JBSteve with wpengine.com

**Comment by Sammitch** [comment]
The fact that the `*-CTR` ciphers were purposefully disabled in the underlying library reeks of dated, reactionary purse-clutching about CTR ciphers.

It should also be noted that Aptana's SFTP implementation also doesn't support anything more advanced than `diffie-hellman-group1-sha1` for key exchange, this algorithm having been dropped from the defaults of recent versions of OpenSSH.

For anyone landing here wondering how to fix it, it is only fixable *server-side*. There is nothing to be done until Aptana releases an update to address this.

For OpenSSH server you need to add `diffie-hellman-group1-sha1` to `KexAlgorithms` and `aes256-cbc` to `Ciphers` in your sshd_config. Refer to the manpage for sshd_config for the defaults for these values if they're not already set explicitly.

**Comment by narsene** [comment]
Having the same issue with gandi.net

**Comment by ParkerD13** [comment]
Same issue here with HostEurope since their recent updates. Their support states "[...]these ciphers represent a security risk which is why it is not possible to reactivate them. You can contact the developers of Aptana on GitHub[...]" with a link to this site. 
So 1. thanks for this great IDE! and 2. scince this post is from Oct 2015 and this relating post https://github.com/aptana/studio3/issues/367 shows no progress either, does anyone know a work around by now?

**Comment by wrossmann** [comment]
Yep, I solved this problem by switching to Eclipse a day or so after my post. No regrets.

**Comment by narsene** [comment]
Same for me i switched to sublime text...

2016-09-23 9:35 GMT+02:00 Wade Rossmann <notifications@github.com>:

**Comment by ParkerD13** [comment]
@ingo I don't wanna switch. I got quite used to aptana and I'll wait for a fix.

**Comment by sgtcoolguy** [comment]
Likely relates to #378 and #334 

I just pushed an updated version of our 3rd-party dependency library for ftp/ftps/sftp support. I believe the builds containing it have gone out to the nightly update sites.

**Comment by lottabliss** [comment]
Dreamhost said that my account won't let me access the "root" to change the cipher.  Is there a fix that I can do from Aptana Studio itself?, I know you wrote stuff about ciphers, but I can't figure how to change the code and could Aptana just change the cipher in the next software release? Thank you.
